rust/pattern/constraint/thing_constraint.rs

```diff
@@ -340,11 +297,11 @@ impl From<RolePlayerConstraint> for RelationConstraint {
     }
 }
 
-impl Display for RelationConstraint {
+impl fmt::Display for RelationConstraint {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_char('(')?;
-        write_joined!(f, self.role_players, ", ")?;
-        f.write_char(')')
+        f.write_str("(")?;
+        write_joined!(f, ", ", self.role_players)?;
+        f.write_str(")")
```

Comment:
`write_char` requires importing a trait; `write_str` has no such prerequisites. Cleaner this way, and seems to be in line with the way major crates do `Display`.